src/perception/vision_pipeline/vision_pipeline/node_detector.py

```diff
@@ -230,15 +168,114 @@ def callback(self, colour_msg: Image, colour_camera_info_msg: CameraInfo, depth_
+    rclpy.spin(detector_node)
+    rclpy.shutdown()
+
+
+## PyTorch inference
+def main_torch(args=None):
+    from .torch_inference import torch_init, infer
+    
+    # loading Pytorch model
+    MODEL_PATH = os.path.join(get_package_share_directory("vision_pipeline"), "models", "YBV2.pt")
+    REPO_PATH = os.path.join(get_package_share_directory("vision_pipeline"), "yolov5")
+    CONFIDENCE = 0.40 # higher = tighter filter 
+    model = torch_init(CONFIDENCE, MODEL_PATH, REPO_PATH)
+
+    def get_torch_bounding_boxes(colour_frame: np.ndarray) -> List[Tuple[Rect, ConeMsgColour, Colour]]:  # bbox, msg colour, display colour
+        bounding_boxes: List[Tuple[Rect, ConeMsgColour, Colour]] = []
+        data = infer(colour_frame, model)
+        
+        for i in range(len(data.index)): 
+            cone_colour = int(data.iloc[i, 5]) # locates object i, class ID at index 5
+            bounding_box = Rect(
+                int(data.xmin[i]),
+                int(data.ymin[i]),
+                int(data.xmax[i]-data.xmin[i]),
+                int(data.ymax[i]-data.ymin[i]),
+            )
+            bounding_boxes.append((bounding_box, cone_colour, CONE_DISPLAY_PARAMETERS[cone_colour]))
+        return bounding_boxes
 
-def main(args=None):
     rclpy.init(args=args)
+    mode = 1
+    detector_node = DetectorNode(mode, get_torch_bounding_boxes)
+    rclpy.spin(detector_node)
+    rclpy.shutdown()
 
-    detector_node = DetectorNode()
 
+## TensorRT inference
+def main_trt(args=None):
+    from .trt_inference import TensorWrapper
+
+    # loading TensorRT engine
+    ENGINE_PATH = os.path.join(get_package_share_directory("vision_pipeline"), "models", "YBV2.engine")
+    PLUGIN_PATH = os.path.join(get_package_share_directory("vision_pipeline"), "models", "libplugins.so")
+    CONFIDENCE = 0.30 # higher = tighter filter 
+    trt_wrapper = TensorWrapper(ENGINE_PATH, PLUGIN_PATH, CONFIDENCE)
+
+    def get_trt_bounding_boxes(colour_frame: np.ndarray) -> List[Tuple[Rect, ConeMsgColour, Colour]]:  # bbox, msg colour, display colour
+        bounding_boxes: List[Tuple[Rect, ConeMsgColour, Colour]] = []
+
+        result_boxes, result_scores, result_classid = trt_wrapper.infer(colour_frame)
+        # Draw rectangles and labels on the original image
+        for i, box in enumerate(result_boxes):
+            cone_colour = int(result_classid[i])
+            bounding_box = Rect(
+                int(box[0]),
+                int(box[1]),
+                int(box[2]-box[0]),
+                int(box[3]-box[1]),
+            )
+            print(box, cone_colour)
+            bounding_boxes.append((bounding_box, cone_colour, CONE_DISPLAY_PARAMETERS[cone_colour]))
+        return bounding_boxes
+
+    rclpy.init(args=args)
+    mode = 2
+    detector_node = DetectorNode(mode, get_trt_bounding_boxes)
     rclpy.spin(detector_node)
     rclpy.shutdown()
 
 
-if __name__ == '__main__':
-    main()
+if __name__ == '__main_cv2__':
```

Comment:
does this actually do anything? Can probably just delete these

Reply:
yeah they arent needed. the setup.py path specifies the function to start, not the name